lib/services/crawler/crawler.go

```diff
@@ -0,0 +1,153 @@
+package crawler
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"gorm.io/gorm"
+	"net/url"
+	"strings"
+
+	"github.com/markgravity/golang-ic/helpers/log"
+	"github.com/markgravity/golang-ic/lib/models"
+
+	"github.com/gocolly/colly/v2"
+	"github.com/gocolly/colly/v2/extensions"
+)
+
+type Crawler struct {
+	DB        *gorm.DB
+	Keyword   *models.Keyword
+	Collector *colly.Collector
+
+	parsingResult *ParsingResult
+}
+
+type ParsingResult struct {
+	HTMLCode        string
+	NonAdwordLinks  []string
+	AdwordLinks     []string
+	ShopAdwordLinks []string
+}
+
+var selectors = map[string][]string{
+	"nonAds":      {"#search .yuRUbf > a", ".ZINbbc.xpd .kCrYT > a"},
+	"ads":         {"#tads .d5oMvf > a", ".uEierd .ZINbbc a.C8nzq.BmP5tf"},
```

Comment:
`d5oMvf`, `ZINbbc`, `C8nzq`, `BmP5tf`, ... looks like they are random generated IDs. I'm not sure if they are reliable filters... 

Should we look more at the `data-` attributes?

<img width="802" alt="image" src="https://github.com/markgravity/golang-ic/assets/77609814/fdac443a-2ca0-4a86-9498-25001a0e054c">

Reply:
@malparty The point is we don't even have this HTML code (on your picture) from the response so these selectors don't have a chance to run 😅

Reply:
@markgravity Let's not implement this and create a new task for ads link as you don't have results from Google

Reply:
Agree, if we don't have ads yet, let's remove `{"#tads .d5oMvf > a", ".uEierd .ZINbbc a.C8nzq.BmP5tf"},` code if it's not implemented yet.